Dockerfile

```
@@ -1,2 +1,45 @@
+FROM fedora AS build
+
+RUN dnf install -y libguestfs \
+  libguestfs-tools-c \
+  virt-v2v \
+  libvirt-daemon \
+  libvirt-daemon-config-network
+
+# https://bugzilla.redhat.com/show_bug.cgi?id=1045069
+RUN useradd -ms /bin/bash v2v
+USER v2v
+WORKDIR /home/v2v
+
+# This is required for virt-v2v because neither systemd nor
+# root libvirtd runs, and therefore there is no virbr0, and
+# therefore virt-v2v cannot set up the network through libvirt.
+ENV LIBGUESTFS_BACKEND direct
+
+# S3 download via HTTPS will be unauthenticated (uses EC2 node role)
+# If needed, download outside of Dockerfile and pass --build-arg=image.ova
+ARG OVA=https://naps-dev-artifacts.s3.amazonaws.com/Capability-SW/MixMode/MixMode-8.14-export-i-0e80aaca99d863be5.ova
+ADD $OVA disk.ova
+
+# Convert OVA to qcow2 with virt-v2v, then edit the VM with virt-customize
+# Cannot use virt-customize --edit, VM doesn't have perl, use sed instead
+# Weird timezone issues, setting Acquire::Max-FutureTime=86400 lets the time be out of sync by 24hr
+# Need to delete Ubuntu installer provided cfg files to re-enable cloud-init
+#  See: https://learn.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-ubuntu
+# Default to dhcp netplan, with low 00 priority
+# Bash $'\n' syntax is used to include newlines in the Dockerfile
+RUN \
+  virt-v2v \
+  -i ova "disk.ova" \
+  -o local -of qcow2 -os "$PWD" -on "disk" && \
+  mv "disk-sda" "disk.qcow2" && \
+  virt-customize \
+  -a "disk.qcow2" \
+  --run-command 'yum install -y cloud-init qemu-guest-agent' \
+  --run-command 'cloud-init clean --logs --seed'  \
+  --append-line '/etc/default/grub:GRUB_TERMINAL=serial' \
+  --append-line '/etc/default/grub:GRUB_SERIAL_COMMAND="serial --speed=115200 --unit=0 --word=8 --parity=no --stop=1"' \
+  --run-command 'grub2-mkconfig -o /boot/grub2/grub.cfg'
```

Not a serious issue, but some extra indentation makes this easier to read:
```suggestion
RUN \
  virt-v2v \
    -i ova "disk.ova" \
    -o local -of qcow2 -os "$PWD" -on "disk" && \
  mv "disk-sda" "disk.qcow2" && \
  virt-customize \
    -a "disk.qcow2" \
    --run-command 'yum install -y cloud-init qemu-guest-agent' \
    --run-command 'cloud-init clean --logs --seed'  \
    --append-line '/etc/default/grub:GRUB_TERMINAL=serial' \
    --append-line '/etc/default/grub:GRUB_SERIAL_COMMAND="serial --speed=115200 --unit=0 --word=8 --parity=no --stop=1"' \
    --run-command 'grub2-mkconfig -o /boot/grub2/grub.cfg'
```